File: baekjoon/javascript/1620.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", line => {
  input.push(line);
}).on("close", () => {
  const [N, M] = input[0].split(" ");
  const nameList = [];
  const nameMap = new Map();
  for (let i = 1; i <= N; i++) {
    nameList.push(input[i]);
    nameMap.set(input[i], i);
  }
  const result = [];
  for (let i = +N + 1; i < input.length; i++) {
    if (!isNaN(input[i])) result.push(nameList[+input[i] - 1]);
    else result.push(nameMap.get(input[i]));
  }
  console.log(result.join("\n"));
});

File: baekjoon/python/1620.py
import sys; input = sys.stdin.readline

N, M = map(int, input().split())
encyclopedia_list = []
encyclopedia_dict = {}
for i in (range(N)):
    name = input().rstrip()
    encyclopedia_list.append(name)
    encyclopedia_dict[name] = i+1

result = []
for _ in range(M):
    ans = input().rstrip()
    result.append(
        encyclopedia_list[int(ans)-1] 
        if ans.isdigit() 
        else encyclopedia_dict[ans]
    )

print(*result, sep="\n")